MekHQ/src/mekhq/campaign/parts/Refit.java

```diff
@@ -1433,20 +1432,37 @@ else if (part instanceof MissingPart) {
                 suit.setUnit(oldUnit);
             }
         }
+        int expectedHeatSinkParts = 0;
+        if ((newEntity instanceof Aero) && ((newEntity.getEntityType() &
+                (Entity.ETYPE_CONV_FIGHTER | Entity.ETYPE_SMALL_CRAFT | Entity.ETYPE_JUMPSHIP)) == 0)) {
+            // Only Aerospace Fighters are expected to have heat sink parts (Mechs handled separately)
+            // SmallCraft, Dropship, Jumpship, Warship, and SpaceStation use SpacecraftCoolingSystem instead
+            expectedHeatSinkParts = ((Aero)newEntity).getHeatSinks() - ((Aero)newEntity).getPodHeatSinks() -
+                    untrackedHeatSinkCount(newEntity);
+        }
         for(int pid : newUnitParts) {
             Part part = oldUnit.getCampaign().getPart(pid);
             if(null == part) {
                 MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
                         "part with id " + pid + " not found for refit of " + getDesc()); //$NON-NLS-1$
                 continue;
             }
-            if (((part instanceof HeatSink) || (part instanceof AeroHeatSink)) &&
-                    ((newEntity instanceof Tank) || (newEntity instanceof Aero))) {
-                // Tanks and Aeros do not track heat sink parts (Mechs handled later)
+            if (part instanceof HeatSink && newEntity instanceof Tank) {
```

Comment:
What about conventional fighters? Those behave the same way as tanks do as I recall.

Reply:
ConvFighters (and all other subclasses of Aero) should enter line 1457 with expectedHeatSinkParts = 0, and have all heat sink parts removed (except OmniPodded ones, but they'll only be present for Aero Fighters).

For readability, I could restore the check for ConvFighters back to line 1450, but I'd have to restore the check for AeroHeatSink as well. Either way works - let me know what you prefer.

Reply:
I don't think it's necessary, let's leave it as is.